orc8r/cloud/go/services/bootstrapper/servicers/registration/registration.go

```diff
@@ -0,0 +1,116 @@
+package registration
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-openapi/strfmt"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"magma/orc8r/cloud/go/orc8r"
+	"magma/orc8r/cloud/go/serdes"
+	"magma/orc8r/cloud/go/services/bootstrapper"
+	"magma/orc8r/cloud/go/services/device"
+	"magma/orc8r/cloud/go/services/orchestrator/obsidian/models"
+	"magma/orc8r/cloud/go/services/tenants"
+	"magma/orc8r/lib/go/protos"
+)
+
+// RegistrationService is public for ease of testing and mocking out functions
+type RegistrationService struct{
+	GetGatewayDeviceInfo func(ctx context.Context, token string) (*protos.GatewayDeviceInfo, error)
+	RegisterDevice func(deviceInfo protos.GatewayDeviceInfo, hwid *protos.AccessGatewayID, challengeKey *protos.ChallengeKey) error
+	GetControlProxy func(networkID string) (string, error)
+}
+
+func NewRegistrationServicer() (protos.RegistrationServer) {
+	return &RegistrationService{
+		GetGatewayDeviceInfo: bootstrapper.GetGatewayDeviceInfo,
+		RegisterDevice: RegisterDevice,
+		GetControlProxy: GetControlProxy,
+	}
+}
+
```

Comment:
quick input would be appreciated :^)

Can take look here and also test file orc8r/cloud/go/services/bootstrapper/servicers/registration/registration_test.go to see how they play together!